### PR TITLE
Add wait profiler

### LINKED ISSUE #295: Add a generic "Launch and wait" function rather than it only being avalible through external tools like profilers
### Role

I'm a mod developer who occasionally need to debug things in prod

### Suggestion

The ability to launch the game and wait without having to also launch a profiler or similar tool

### Benefit

Attaching a remote debugger to debug launch stuff currently requires launching an entirely seperate program to get the "waiting" box, and this would remove that

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

Ive been doing some production debugging, and the inability to pause the game launch to connect my remote debugger without having to launch an extra, unnecessary tool slows me down quite a bit. You also can't always just launch normally because sometimes bugs are so early that the time it takes to switch to your debugger and tell it too connect is too long.

### PR DESCRIPTION
**Pull request by Trial97**
added a profiler that does nothing(praise me :D)
resolves #295 by adding functionality that launches the game but waits until you click the ok button to actually start it, obviously without starting another background program.
And for functionality, it just displays the pid of the started process.

### CODE REVIEW

**Comment by TheKodeToad** [comment]
Maybe it should just be "Generic" or something

**Comment by Trial97** [comment]
> Maybe it should just be "Generic" or something

Does "Generic Profiler" sounds good then?

**Comment by TheKodeToad** [comment]
It's already in the profiler menu :woman_shrugging:

**Comment by Trial97** [comment]
I'm confused: what do you mean is in profiler menu ?
Obviously is there but the name is still Wait Profiler not Generic Profiler.
So should I change it?

**Comment by TheKodeToad** [comment]
What I mean is you don't need extra clarification that it's a Profiler

File: launcher/tools/GenericProfiler.cpp [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    void beginProfilingImpl(shared_qobject_ptr<LaunchTask> process);
+};
+
+GenericProfiler::GenericProfiler(SettingsObjectPtr settings, InstancePtr instance, QObject* parent)
+    : BaseProfiler(settings, instance, parent)
+{}
+
+void GenericProfiler::beginProfilingImpl(shared_qobject_ptr<LaunchTask> process)
+{
+    emit readyToLaunch(tr("Started process: %1").arg(process->pid()));
```
Comment (TheKodeToad) [inline launcher/tools/GenericProfiler.cpp:39]: isn't this already shown?

Comment (Trial97) [reply launcher/tools/GenericProfiler.cpp:39]: If you mean the text no the method is a pure virtual one. So I need to overload it

Comment (TheKodeToad) [reply launcher/tools/GenericProfiler.cpp:39]: I haven't looked at in detail. but i don't think this profiler needs to print anything so maybe you should find a way to.. not.

Comment (Trial97) [reply launcher/tools/GenericProfiler.cpp:39]: It should in order to let the user open its own profiler/debugger to attach to the process

Comment (TheKodeToad) [reply launcher/tools/GenericProfiler.cpp:39]: image

Comment (Trial97) [reply launcher/tools/GenericProfiler.cpp:39]: Yea, I know that it's in logs what I mean is that I still need a dialog box to display to the user. 
Why not use that to also display the process ID?
If I remove that the message will be something like this:
```suggestion
    emit readyToLaunch(tr("Process started"));
```

Comment (TheKodeToad) [reply launcher/tools/GenericProfiler.cpp:39]: Oh, nvm lol

**Review by TheKodeToad — APPROVED** [review]

**Review by DioEgizio — APPROVED** [review]

**Pull request merged**